Add support for running custom actions when modifiers are held. (#4795)
qmk/qmk_firmware: docs/ref_functions.md

  // do something if 100ms or more have passed
}
```
## Custom modified keycode handling

> Sometimes you might want to send a specific keycode or execute a specific macro

Reviewer: Could this be placed all on one block, not a bunch of separate lines?

when some modifiers and a key are pressed. For example let's say that your
keymap binds a shifted keycode like question mark (?) on a qwerty keyboard to a
key. When pressed, qmk will send these events: shift pressed, slash (/) pressed,
slash released, shift released. But then what happens when pressing shift and

> that key ? Qmk will send shitf pressed, slash pressed, slash released and shift

Reviewer: ```suggestion
that key? QMK Firmware will send shift pressed, slash pressed, slash released and shift
```

released. The OS won't be able to distinguish between the two and you might just

> have lost a usefull keycombo on your shiny new keymap. This is  where the

Reviewer: ```suggestion
have lost a useful key combo on your shiny new keymap. This is where the
```

> `custom_keycode_on_modifiers` function comes in. It can be used to have qmk send

Reviewer: ```suggestion
`custom_keycode_on_modifiers` function comes in. It can be used to have QMK Firmware send
```

another keycode when some modifiers are pressed. Its prototype is

```c
bool custom_keycode_on_modifiers(uint8_t modifier_mask, uint8_t layer, keyrecord_t *record, uint16_t custom_keycode)
```

> The first argument is the modifiers that must be held for the custom keycode to

Reviewer: I think this may be better if you present a use case for this, first, and then how to do it in the code. 

Also, again, could you make this block one line?

Author: I'm not following you here. The uses case is the first thing I explain in the new documentation on line 120. Is that not enough ? Or am I missing something ?

Author: ping

be sent. Use the `MOD_BIT(keycode)` macro to create it. The second argument is
the layer that must be active for the custom keycode to be sent. Use -1 if it
must be available on all layers. The third argument is the key record being
processed and the last one is the custom keycode you want to set when the
modifiers and layer conditions are met. For example:

```c
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
  switch(keycode) {
    case KC_QUESTION:
      return custom_keycode_on_modifiers(MOD_BIT(KC_LSHIFT), -1, record,  KC_EXCLAIM);
  }

  return true;
}

```

When ? is pressed and left shift is hold, on any layer, send !. You can combine
multiple modifiers like this:

```c

> MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_LCTRL())

Reviewer: ```suggestion
MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_LCTRL)
```
